[RM Ch07] Chapter7 updater

d-j-eagle opened the pull request:
Change to the reference model doco - cleanup and added to back-end and front-end descriptions. Still trying to keep this general.

Also, removed some sections that are probably best covered in the reference architecture section.

rabiabdel commented:
@glimonta can you review this please! since you had a similar PR few days ago.

rabiabdel commented:
@d-j-eagle this has conflicts due to a previous PR that has been approved and merged. would you mind please resolving the conflicts above?

rabi-abdel commented:
@d-j-eagle it is still showing "This branch has conflicts that must be resolved" you can do this by clicking "Resolve conflicts" and use the GitHub GUI to do that.

d-j-eagle commented:
> @d-j-eagle it is still showing "This branch has conflicts that must be resolved" you can do this by clicking "Resolve conflicts" and use the GitHub GUI to do that.

Sorry @rabiabdel, I'm not an experienced Git user. Not sure if this worked...suspect I did something wrong as I appear to have an issue with 'labeler'?

rabi-abdel commented:
@d-j-eagle I think that is OK. you have fixed it. Thanks :) ignore the labeller as this is due to the fact that you used local fork not a branch for the PR. it is ok.

@ASawwaf can you review please

pgoyal01 commented:
@d-j-eagle 

In general, acronyms used first time should be expanded at least those that may not be in general use ("2FA" for two-factor authentication). 

7.4.3. would we need secure access controls for Tenants?
         Should "strong separation between tenants" be about tenant isolation (or separation needs to be defined)
        Can we have a single "Secure APIs and Authenticated access to APIs for all NFVI services"?
        Should we have "Authorized platform hardware, including configuration, change control" and ""Authorized platform software change control"
        
7.4.4. What does "Each workload network will have a separate/specific security assessment?" is this for tenant networks?

In the following is the reference restricted to workloads as in "Workload owner owns workloads security certification process" (BTW should this be "Workload owner owns security the workload"):
- Workload owner owns security certification process
- Workload owner owns design change process
- Workload owner owns software update process

Section 7.5.1. I think "carriage network" needs to be defined.
"No login to root on any platform systems" -- what is included in "platform systems"?

"RBAC must apply for all systems access" should RBAC be applicable to all sections not just "platform Security? Should there be guidance on the granularity of the roles or should a set of roles be defined with their access rights.

Is the section on 7.6 Workload Security is yet to be developed?

7.7. Vendor Responsibility and 7.9 VNF Vendor Responsibility: should 7.7 be qualified as to the nature of the vendor?

7.8.2. ".. VNF images can be cryptographically signed and verified during launch time." can this be clarified so that images come with a signature (specified say in 7.9) -- to me this reads as if both the "signing" and the "verification" is happening at launch time.

glimonta commented on doc/ref_model/chapters/chapter07.md:
```diff
@@ -16,13 +16,12 @@
 * [7.5 Platform Security](#7.5)
-  * [7.5.1 Platform Security Assumption](#7.5.1)
-  * [7.5.2 Platform ‘back-end’ access security](#7.5.2)
+  * [7.5.1 General Platform Security.](#7.5.1)
```
Remove periods from headings.

glimonta commented on doc/ref_model/chapters/chapter07.md:
```diff
@@ -16,13 +16,12 @@
 * [7.5 Platform Security](#7.5)
-  * [7.5.1 Platform Security Assumption](#7.5.1)
-  * [7.5.2 Platform ‘back-end’ access security](#7.5.2)
+  * [7.5.1 General Platform Security.](#7.5.1)
```
Remove periods from headings.

rabi-abdel replied:
I did this on behalf of @d-j-eagle

glimonta commented on doc/ref_model/chapters/chapter07.md:
```diff
@@ -225,259 +224,57 @@ The platform supports the workload, and in effect controls access to the workloa
-    connection and provide additional security controls).*
-
-<a name="7.5.4"></a>
-## 7.5.4 Platform services
```
Is it so that these sections that are removed are being moved to the RA document?
Is that part of a different PR?

rabi-abdel replied:
@glimonta correct. Chapter 9 I believe in RA1. Please resolve if you are happy.

rabi-abdel commented:
@d-j-eagle can you please address the few points above so we can get this approved and merged.

rabi-abdel commented:
@d-j-eagle are you able to resolve comments above?

d-j-eagle commented:
Hi Rabi,

Sorry, caught up with other matters. Will look at these items and resolve over the next 24 hours,

Cheers,
Damian

Sent from my Samsung Mobile on the Telstra Mobile Network
Get Outlook for Android<https://aka.ms/ghei36>

________________________________
From: Rabi Abdel <notifications@github.com>
Sent: Monday, December 2, 2019 10:24:18 PM
To: cntt-n/CNTT <CNTT@noreply.github.com>
Cc: Eagle, Damian <Damian.J.Eagle@team.telstra.com>; Mention <mention@noreply.github.com>
Subject: Re: [cntt-n/CNTT] [RM Ch07] Chapter7 updater (#650)


[External Email] This email was sent from outside the organisation – be cautious, particularly with links and attachments.

@d-j-eagle<https://github.com/d-j-eagle> are you able to resolve comments above?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cntt-n/CNTT/pull/650?email_source=notifications&email_token=AM647IMRHKJ3V7ZEOPKJXZLQWTV6FA5CNFSM4JO5NBYKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFTFM7Y#issuecomment-560354943>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AM647IP3I4CW2PH3HHJEMBDQWTV6FANCNFSM4JO5NBYA>.

d-j-eagle commented:
@pgoyal01 - thank you for your review.  Unfortunately I've been caught up and haven't had a chance to respond to you ... See my comments below

> In general, acronyms used first time should be expanded at least those that may not be in general use ("2FA" for two-factor authentication).
**DE: I've expanded acronyms that are particular to this section. Some such as VNF (will be referenced in the previous sections, and others such as protocol names (SSH for example) will be better known as their acronym.**
> 
> 7.4.3. would we need secure access controls for Tenants?
**DE: Tenants may connect to tenant services via their workload interface, and so it's difficult to make this a 'requirement'.**
> Should "strong separation between tenants" be about tenant isolation (or separation needs to be defined)
**DE: This is really about specifically stating that a main intent of tenant isolation is separation from other tenants. I've added some extra words to make this clearer.**
> Can we have a single "Secure APIs and Authenticated access to APIs for all NFVI services"?
**DE: I don't think this will be a correct statement. The requirement for 'extra' security (i.e. additional authentication steps) for example will be dependent on whether the tenant can be trusted. If the tenant originates from the tenant workload network, then we can trust the network connection. The tenant would likely want us to assume that the workload-to-infrastructure path is trusted. Therefore we can't mandate this requirement. In some installations where this access does not exist, then I can see that it could be mandated.**
> Should we have "Authorized platform hardware, including configuration, change control" and ""Authorized platform software change control"
**DE: You are proposing the concept of trusted h/w?  Although I understand the concept, it may not exist in all instances as I believe these are likely to be virtual/software and not associated with specific hardware. Again, it would be hard to mandate this?**
> 
> 7.4.4. What does "Each workload network will have a separate/specific security assessment?" is this for tenant networks?
**DE: Improved the wording of this one. It's basically saying that each workload's security assessment is specific to that workload, and cannot inherit the security accreditation of the platform.**
> 
> In the following is the reference restricted to workloads as in "Workload owner owns workloads security certification process" (BTW should this be "Workload owner owns security the workload"):
> 
> * Workload owner owns security certification process
> * Workload owner owns design change process
> * Workload owner owns software update process
**DE: Yes, this is the case - I've updated the wording to reflect 'workload**
>
> Section 7.5.1. I think "carriage network" needs to be defined.
DE: OK, I've added more clarification around carriage networks

> "No login to root on any platform systems" -- what is included in "platform systems"?
**DE: Have added an extended definition for platform systems.**
> 
> "RBAC must apply for all systems access" should RBAC be applicable to all sections not just "platform Security? Should there be guidance on the granularity of the roles or should a set of roles be defined with their access rights.
**DE: OK, I've added a qualifier to make this just platform systems. I'm thinking that the extra qualifier around further granularity would be in the architecture section(s)?**
> 
> Is the section on 7.6 Workload Security is yet to be developed?
**DE: Yes, this section has not yet been completed :-(**
> 
> 7.7. Vendor Responsibility and 7.9 VNF Vendor Responsibility: should 7.7 be qualified as to the nature of the vendor?
**DE: Not 100% sure of the thinking here, but it sounds reasonable that responsibility would relate to function. I'll have to refer to @mbustame for comment on these sections. **
> 
> 7.8.2. ".. VNF images can be cryptographically signed and verified during launch time." can this be clarified so that images come with a signature (specified say in 7.9) -- to me this reads as if both the "signing" and the "verification" is happening at launch time.
**DE: Not my inclusion. I'll have to refer to @mbustame for details here on his thinking. **

pgoyal01 commented on doc/ref_model/chapters/chapter07.md:
```diff
@@ -227,28 +227,28 @@ The platform supports the workload, and in effect controls access to the workloa
-* **FRONT-END ACCESS CONTROLS** - Protects the platform from malicious carriage network access, and provides connectivity for specific VNFs to specific carriage networks (usually sub, or virtual networks).
-* **TENANT ACCESS CONTROLS** - Provides apropriate tenant access controls to specific platform services, and tenant workloads - including RBAC, authentication controls as approriate for the access arrangement, and API gateways for automated/script-driven processes.
+* **BACK-END ACCESS CONTROLS** - Data Centre Operations access to the platform, and subsequently, workloads. Typically stronger authentication requirements such as (Two-Factor Authentication) 2FA, and using technologies such as Role-Based Access Control (RBAC) and encryption. Application Programming Interface (API) gateways may be required for automated/script-driven processes.
+* **FRONT-END ACCESS CONTROLS** - Protects the platform from malicious carriage network access, and provides connectivity for specific VNFs to specific carriage networks. Carriage networks being those that are provided as public networks and operated by carriers, and in this case with interfaces that are usually sub, or virtual networks.
```
@d-j-eagle Should "carriage network" be generalised and not limited to "... provided as public networks and operated by carriers ..."? Maybe, "... provided as public networks by (cloud) operators ..." -- "cloud" is optional and so shown in parentheses.

d-j-eagle replied:
@pgoyal01 - OK, fair point - happy to remove the 'carrier'.  I also removed the 'cloud' to be consistent with being generalist, and added 'network operator' to be more accurate.     So it becomes "provided as public networks by network operators".  Is this OK?

rabiabdel commented:
@pgoyal01 @glimonta can you confirm and mark your input as resolved if you are happy! if no respond received within 48 hours, this will be approved and merged.
